spec/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context_spec.rb

```diff
@@ -0,0 +1,90 @@
+      '1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2V'\
+      'ydmljZS1hY2NvdW50LnVpZCI6IjYxNWU2MWRkLTM'\
+      'yMTMtMTFlNi1iMzg3LTAwMWE0YTE2OTc5OSIsInN'\
+      '1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDptYW5'\
+      'hZ2VtZW50LWluZnJhOm1hbmFnZW1lbnQtYWRtaW4'\
+      'ifQ.lnv42HrZ3hujxG8kIovb89XiOVdJEY83FU6h'\
+      'sfA5RFK05gMI8vqBStJmFYpCTRScrBXkljajDKX7'\
+      'OT83mhZ6JoWkfJtDAul4NE7tV1NL_yr1IT-YwNY0'\
+      'eCD_BjvJw-H_JdEDo0ivyByAHnfJOB7GK27a6DtO'\
+      'Yu4aR5eqlV7kQb3zusgNcAScEeYJcVDHWbGgr0hB'\
+      'c1E7Zc4EvvjZWOmv_S7H5RxpZfXkPvOr20-rEpaj'\
+      '0RJ2hs7nxZv4ahuSPUlwzSQdRj_BQss2iGy7mbSp'\
+      'gBbHTTagOHwhrZmMSgwfwpzPAiFij42xF61HCK_Z'\
+      'LrDEZ7_HDfHAXRa0pHEbh4D2IQ'
+
+    @ems = FactoryGirl.create(
+      :ems_openshift,
+      :name                      => 'QaLab',
+      :connection_configurations => [{:endpoint       => {:role     => :default,
+                                                          :hostname => hostname,
+                                                          :port     => "8443"},
+                                      :authentication => {:role     => :bearer,
+                                                          :auth_key => token,
+                                                          :userid   => "_"}},
+                                     {:endpoint       => {:role     => :hawkular,
+                                                          :hostname => hostname_hawkular,
+                                                          :port     => "443"},
+                                      :authentication => {:role     => :hawkular,
+                                                          :auth_key => token,
+                                                          :userid   => "_"}}])
+  end
+
+  it "will read hawkular status (does not require authentication)" do
+    VCR.use_cassette("#{described_class.name.underscore}_status") do # , :record => :new_episodes) do
+      EmsRefresh.refresh(@ems)
+      context = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
+        @ems.container_nodes.last, 1_466_505_130, 1_466_508_726, 20)
+
+      metrics = {"MetricsService"         => "STARTED",
+                 "Implementation-Version" => "0.8.1.Final-redhat-1",
+                 "Built-From-Git-SHA1"    => "f8718499684ef4b9beb456667fa7da4786e7fadd"}
+
+      data = context.hawkular_client.http_get('/status')
+
+      expect(data).to eq(metrics)
+    end
+  end
+
+  it "will read hawkular system metrics (requires authentication)" do
+    VCR.use_cassette("#{described_class.name.underscore}_system_metrics") do # , :record => :new_episodes) do
+      EmsRefresh.refresh(@ems)
+      context = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
+        @ems.container_nodes.last, 1_466_505_130, 1_466_508_726, 20)
+
+      data = context.hawkular_client.http_get('/metrics')
```

Comment:
Hi @yaacov a minor thing (non blocker from my side).
I see that the containers provider basically process itself the metrics data and it uses just to proxy the call.
In middleware provider we are trying to put an abstract layer to this like this one:
https://github.com/hawkular/hawkular-client-ruby/blob/master/lib/hawkular/metrics/metric_api.rb

Not for this one, but perhaps in future is useful this approach for you too. For example, I'm thinking that some processing work can be simplified (and maintained) inside of the hawkular-client.

Apart of this comment, I've fetched the PR and I can validate that it is using the correct version of hawkular-client and container tests are green on my side.

Probably travis is giving non related issues, I had some of that this morning too.